tests/Chat/ChatSmokeTests.cs

```diff
@@ -585,4 +586,29 @@ public void SerializeMessagesWithNullProperties()
         FunctionChatMessage deserializedMessage = ModelReaderWriter.Read<FunctionChatMessage>(serializedMessage);
 #pragma warning restore
     }
+
+    [Test]
+    public void TopLevelClientOptionsPersistence()
+    {
+        MockPipelineTransport mockTransport = new(BinaryData.FromString("{}"), BinaryData.FromString("{}"));
+        OpenAIClientOptions options = new()
+        {
+            Transport = mockTransport,
+            Endpoint = new Uri("https://api.openai.com/expected/test/endpoint"),
```

Comment:
Suggest adding to the test changes in the base address as well not just the path.

```suggestion
            Endpoint = new Uri("https://my.custom.com/expected/test/endpoint"),
```